### ISSUE TITLE
Tweak mode not working

### ISSUE THREAD

**Issue by xphilup2** [processing/processing#5805]
## Description
Always get message when using tweak mode

## Expected Behavior
Be able to change values on the fly.

## Current Behavior
java.lang.ClassCastException: processing.app.RunnerListenerEdtAdapter cannot be cast to processing.mode.java.JavaEditor
	at processing.mode.java.JavaMode.handleTweak(JavaMode.java:145)
	at processing.mode.java.JavaEditor.lambda$0(JavaEditor.java:1101)
	at java.lang.Thread.run(Thread.java:748)

## Steps to Reproduce

1.
2.
3.

## Your Environment
Ubuntu 18:04

**Comment by marxmit** [comment]
This also happened to me in all 3.5 versions so far..
Same erro mesage as above.
I am in 64 bit Windows 10.

**Comment by danieltorrer** [comment]
Same issue in 
win 10 64
processing 3.5.3

**Comment by GiselleLlamas** [comment]
Same issue on: Processing 3.5.3
OS: Windows 10
See screenshot below:
![image](https://user-images.githubusercontent.com/25014200/54710324-2f8b2780-4b26-11e9-86dd-40b0a97dba5e.png)

**Comment by Krammy** [comment]
Same issue on Windows 10; Tweak mode shows the above message, and doesn't work.

**Comment by tank-t-bird** [comment]
Same issue on MacOS Mojave 10.14.3 with Processing 3.5.3

It won't run the program and gives this error:

java.lang.ClassCastException: processing.app.RunnerListenerEdtAdapter cannot be cast to processing.mode.java.JavaEditor
	at processing.mode.java.JavaMode.handleTweak(JavaMode.java:145)
	at processing.mode.java.JavaEditor.lambda$0(JavaEditor.java:1101)
	at java.lang.Thread.run(Thread.java:748)

**Comment by KorySchneider** [comment]
Same issue on Windows 10 Processing 3.5.3 and Ubuntu 18 Processing 3.5.3.

Error message from Windows 10:
```
java.lang.ClassCastException: processing.app.RunnerListenerEdtAdapter cannot be cast to processing.mode.java.JavaEditor
	at processing.mode.java.JavaMode.handleTweak(JavaMode.java:145)
	at processing.mode.java.JavaEditor.lambda$0(JavaEditor.java:1101)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by dzaima** [comment]
Seems that this is caused by [this](https://github.com/processing/processing/commit/ec8de86dbf7a62d857f9ef4211dc325267218bee#diff-1b2be6a19d6be381b1f87099ec8f7bd5L1099) change. Reverting that makes tweak mode work, but here's probably a good reason the change was made and should probably be handled better than just a revert.

**Comment by beladfo** [comment]
mac 10.14.4
Processing 3.5.3
java.lang.ClassCastException: processing.app.RunnerListenerEdtAdapter cannot be cast to processing.mode.java.JavaEditor
	at processing.mode.java.JavaMode.handleTweak(JavaMode.java:145)
	at processing.mode.java.JavaEditor.lambda$0(JavaEditor.java:1101)
	at java.lang.Thread.run(Thread.java:748)

**Comment by jeremydouglass** [comment]
@JakubValtar what do you think of @dzaima's guess that #5486 broke Tweak mode? Do you think it would be possible for someone to fix Tweak mode given the changes in 5486, or is the mode not possible going forward, and should it be removed or not install?

**Comment by KorySchneider** [comment]
Would be a shame to remove the tweak mode, it can be very useful in certain contexts.

**Comment by dzaima** [comment]
I've fixed tweak mode on my branch off of sampottingers Java 11 work [here](https://github.com/dzaima/processing/commit/89d5822452caa65248833ff52bb501130ec586a7). The fix for Processing 3 should be pretty much the same, but I'm having trouble testing because of the Oracle Java self-downloading problem. It might have trouble with threads & synchronization, but it works for me ¯\\\_(ツ)\_/¯

**Comment by donausoft** [comment]
Same problem here (Win10 64-bit), same error:
java.lang.ClassCastException: processing.app.RunnerListenerEdtAdapter cannot be cast to processing.mode.java.JavaEditor
	at processing.mode.java.JavaMode.handleTweak(JavaMode.java:145)
	at processing.mode.java.JavaEditor.lambda$0(JavaEditor.java:1101)
	at java.lang.Thread.run(Thread.java:748)

**Comment by GusFermin** [comment]
Hello.
I am having this exact same issue with same errors in console. 
My Environment:
Processing version: 3.5.3
Operating System and OS version: macOS El Capitan 10.11.6

**Comment by galsasson** [comment]
Good news! I have a fix for you here #5909

**Comment by benfry** [comment]
That PR works superficially but ignores the issue that https://github.com/processing/processing/commit/ec8de86dbf7a62d857f9ef4211dc325267218bee#diff-1b2be6a19d6be381b1f87099ec8f7bd5L1099 fixes (see its description at the top), so it's just pushing the problem for later… A proper solution needs to tease out any EDT mess for Tweak in the same way.

**Comment by galsasson** [comment]
I see, but in this case, no AWT methods are being called by ```handleTweak``` from a different thread. This function uses ```SwingUtilities.invokeLater``` when calling editor functions from a different thread (which is similar to ```EventQueue.invokeLater```, I presume).

```handleTweak``` uses the following functions/fields of the editor:
```
- initBaseCode()
- baseCode (string array)
- automateSketch(...)
- initEditorCode(...) **
- updateInterface(...) **
- startTweakMode() **
- stopTweakMode(...) **
```

- ** are being called from a different thread using ```invokeLater```.

I agree the above interface between JavaEditor and JavaMode is not very well defined and structured (to say the least), but creating another adapter will not solve any issues, and I don't think it will make it easier to understand and maintain the code either.

**Comment by benfry** [comment]
Ok, just leaves us with a mess, though—since we've got two ways that the EDT stuff is being handled. I'll just clean it up later. Thanks for looking into it.

**Comment by benfry** [comment]
Now merged for any release later than 3.5.3.
